From 923ab10f11a493f458d8ac069d76da6d56e7b194 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Mon, 19 Sep 2022 14:36:21 +0200
Subject: [PATCH 1/3] soc: qcom: spm: Add ipq8064 CPU data

Add ipq8064 CPU data for boot and non-boot CPU. Load only the WFI
sequence as some SoC seems to be broken with SPC sequence and seems to
be not supported on this platform.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/soc/qcom/spm.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/soc/qcom/spm.c b/drivers/soc/qcom/spm.c
index 484b42b7454e..d822ea6dee38 100644
--- a/drivers/soc/qcom/spm.c
+++ b/drivers/soc/qcom/spm.c
@@ -151,6 +151,30 @@ static const struct spm_reg_data spm_reg_8064_cpu = {
 	.start_index[PM_SLEEP_MODE_SPC] = 2,
 };
 
+/* SPM register data for ipq8064 boot cpu */
+static const struct spm_reg_data spm_reg_ipq8064_boot_cpu = {
+	.reg_offset = spm_reg_offset_v1_1,
+	.spm_cfg = 0x1F,
+	.pmic_dly = 0x03020004,
+	.pmic_data[0] = 0x0084009C,
+	.pmic_data[1] = 0x00A4001C,
+	.seq = { 0x03, 0x0F,
+	},
+	.start_index[PM_SLEEP_MODE_STBY] = 0,
+};
+
+/* SPM register data for ipq8064 non boot cpu */
+static const struct spm_reg_data spm_reg_ipq8064_nonboot_cpu = {
+	.reg_offset = spm_reg_offset_v1_1,
+	.spm_cfg = 0x1F,
+	.pmic_dly = 0x02020004,
+	.pmic_data[0] = 0x0060009C,
+	.pmic_data[1] = 0x0000001C,
+	.seq = { 0x03, 0x0F,
+	},
+	.start_index[PM_SLEEP_MODE_STBY] = 0,
+};
+
 static inline void spm_register_write(struct spm_driver_data *drv,
 					enum spm_reg reg, u32 val)
 {
@@ -221,6 +245,10 @@ static const struct of_device_id spm_match_table[] = {
 	  .data = &spm_reg_8974_8084_cpu },
 	{ .compatible = "qcom,apq8064-saw2-v1.1-cpu",
 	  .data = &spm_reg_8064_cpu },
+	{ .compatible = "qcom,ipq8064-saw2-v1.1-boot-cpu",
+	  .data = &spm_reg_ipq8064_boot_cpu },
+	{ .compatible = "qcom,ipq8064-saw2-v1.1-nonboot-cpu",
+	  .data = &spm_reg_ipq8064_nonboot_cpu },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, spm_match_table);
-- 
2.37.2

